src/renderer/include/vector.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <fmt/format.h>
+
+struct Vec2D {
+	int x;
+	int y;
+};
+
+struct Vec3D {
+	int x;
+	int y;
+	int z;
+};
+
+template <>
+struct fmt::formatter<Vec2D> {
+	static constexpr auto parse(format_parse_context& ctx) -> decltype(ctx.begin()) {
+		return ctx.end();
+	}
```

Comment:
function looks strange, I think we need a comment here

Reply:
For other branch with refactoring 
For clarifying: it just the simplest way to make a type `fmt` compatible without any formatting. Again, it well documented in cppreference and fmt library documentation, so I don't think it is needed to commented here.